metagraph/src/common/vectors/vector_algorithm.hpp

```diff
@@ -109,6 +110,59 @@ inline void unset_bit(uint64_t *v,
     }
 }
 
+inline uint64_t atomic_increment(sdsl::int_vector<> &vector, uint64_t i,
+                                 uint64_t count = 1,
+                                 int memorder = __ATOMIC_SEQ_CST) {
+#ifdef MODE_TI
+    size_t width = vector.width();
+    uint64_t bit_pos = i * width;
+    __uint128_t *limb = &reinterpret_cast<__uint128_t*>(vector.data())[bit_pos >> 7];
+    __uint128_t mask = ((__uint128_t(1) << width) - 1) << (bit_pos & 0x7F);
+    __uint128_t new_val;
+    __uint128_t inc = __uint128_t(count) << (bit_pos & 0x7F);
+    __uint128_t exp_val = *limb;
+    do {
+        new_val = ((exp_val + inc) & mask) | (exp_val & (~mask));
+    } while (!__atomic_compare_exchange(limb, &exp_val, &new_val, true,
+                                        memorder /* order for exchange */,
+                                        __ATOMIC_RELAXED /* order for compare */));
+    return (exp_val & mask) >> (bit_pos & 0x7F);
+#else
+    std::ignore = memorder;
+    static std::mutex mu;
```

Comment:
ideally, this should be a local mutex for vector, in case you write to two or more vectors at the same time.

Reply:
Rather than passing a mutex, I'm thinking of using a builtin function for locking those memory locations.

Reply:
I ended up replacing this with a local mutex since that was the easiest approach.